packages/squiggle-lang/src/rescript/Utility/E.res

```diff
@@ -620,6 +620,28 @@ module A = {
       | Some(o) => o
       | None => []
       }
+    let rec arrSomeToSomeArr = (optionals: array<option<'a>>): option<array<'a>> => {
+      let optionals' = optionals->Belt.List.fromArray
+      switch optionals' {
+      | list{} => []->Some
+      | list{x, ...xs} =>
+        switch x {
+        | Some(_) => xs->Belt.List.toArray->arrSomeToSomeArr
+        | None => None
+        }
+      }
+    }
+    let rec firstSome = (optionals: array<option<'a>>): option<'a> => {
```

Comment:
I feel like you could do this much more simply, without recursion. Like,
``Belt.Array.getBy(array, isSome)``